### PR TITLE
Improve wording on W0402 message

### LINKED ISSUE #6169: Typo in W0402 message
W0402 (`deprecated-module`) error message says "Uses of a deprecated module 'optparse'".

I think it should say "Use of a deprecated module 'optparse'" instead of "Uses". Either "Use", or "Usage". "Uses" seems grammatically incorrect to me. What do you think?

### PR DESCRIPTION
**Pull request by omarandlorraine**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :scroll: Docs          |

## Description

I left the scrolly thing up there since the Issue has the Documentation tag, hope that's the right thing to do.

So I have improved the wording on W0402; it now says "Deprecated module \<module name\>" for consistency and style.

Closes #6169

### CODE REVIEW

**Comment by omarandlorraine** [comment]
Oh pants!

I think commit f4d4107 should not be part of this branch. Should I try and rewrite history or maybe it doesn't matter?

**Comment by omarandlorraine** [comment]
Does anyone know what rstcheck and mypy means and why it's failing the precommit hook?

**Comment by Pierre-Sassoulas** [comment]
Do not take the current pre-commit fail into account, our main branch is broken we'll fix it :)

File: ChangeLog
Comment (DanielNoord) [inline ChangeLog:12]:
```suggestion
* Improved wording of the message of ``deprecated-module``
```

Could you also add this to ``doc/whatsnew/2.14``?

**Review by DanielNoord — CHANGES_REQUESTED** [review]

File: pylint/checkers/imports.py
Comment (DanielNoord) [inline pylint/checkers/imports.py:194]: We will probably need to update some tests as we test for the message of a warning and that changes with this PR.
You can do this by using the auto-updates (the failing tests will show how to do this) or by simplying searching for `Uses of a deprecated module` in our codebase and replacing it with the new message.

Comment (omarandlorraine) [reply pylint/checkers/imports.py:194]: this is resolved by [d14785a](https://github.com/PyCQA/pylint/pull/6387/commits/d14785a5a35b070f55c17faf2c78a3d6b7296997)

now `python ./tests/test_functional.py` shows 726 passed, 24 skipped (why are some tests being skipped?)

Comment (omarandlorraine) [reply pylint/checkers/imports.py:194]: oh hold on, there are more failing tests in other scripts (is there an obvious way to run them all?)

Comment (DanielNoord) [reply pylint/checkers/imports.py:194]: I believe `python tests/test_functional.py --update-functional-output -k "test_functional"` should update them all at once.

Skipped tests are dependent on your environment. Some tests are only run on certain python version or operating systems.

**Comment by DudeNr33** [comment]
> Does anyone know what rstcheck and mypy means and why it's failing the precommit hook?

``rstcheck`` is a [syntax checker for reStructuredText files](https://github.com/myint/rstcheck). It checks all ``*.rst`` files as well as ``Changelog`` (which is written in reStructuredText as well).
``mypy`` is an [optional static type checker](http://mypy-lang.org/) for Python and used to check type conflicts in the code base.

I encourage you to install and activate ``pre-commit`` in your own development environment according to [our docs](https://pylint.pycqa.org/en/latest/development_guide/contribute.html#repository).
This will run all configured checks against the staged changes when you run ``git commit``. 
This not only allows you to fix issues before you push your branch, but also makes it a bit more transparent what the problems are, as you see the tool output directly on your command line.
Last but not least you will quickly learn what the style checks etc. for ``pylint`` expect, and you can write code and documentation directly in a way that the ``pre-commit`` hooks won't complain. :)

**Comment by omarandlorraine** [comment]
> I encourage you to install and activate `pre-commit` in your own development environment according to [our docs](https://pylint.pycqa.org/en/latest/development_guide/contribute.html#repository). 

Oh, that's a great idea and works really well, thanks

**Review by omarandlorraine — COMMENTED** [review]
From what I see, this review needs to be concluded or something before the remaining checks will run.

Sorry if I'm generating a lot of spam, I don't know github that well yet.

**Comment by Pierre-Sassoulas** [comment]
> From what I see, this review needs to be concluded or something before the remaining checks will run.

To prevent bitcoin mining first time contributors need to be approved before the pipeline runs :) You won't have to be approved after this one get merged !

**Review by Pierre-Sassoulas — APPROVED** [review]

**Comment by omarandlorraine** [comment]
Can anyone see if the remaining tests are failing because of an issue with the PR or because of that problem with the main branch?

The precommit passes locally on my machine but there is a "Some checks were not successful" on github and I don't understand. It is worth noting that here I have python 3.9.2 only, and this is presumably why the tests for other versions are not being run.

My working tree is clean and up to date with origin/W0402polish, (origin being my github repo.) and `git grep -i "uses of a deprecated"` returns nothing.

**Comment by DanielNoord** [comment]
> Can anyone see if the remaining tests are failing because of an issue with the PR or because of that problem with the main branch?
> 
> The precommit passes locally on my machine but there is a "Some checks were not successful" on github and I don't understand. It is worth noting that here I have python 3.9.2 only, and this is presumably why the tests for other versions are not being run.
> 
> My working tree is clean and up to date with origin/W0402polish, (origin being my github repo.) and `git grep -i "uses of a deprecated"` returns nothing.

Looking at the test output the expected output for `deprecated_module_redundant` needs to be updated. You can probably do this manually by changing the message in `deprecated_module_redundant.txt` which can be found in the `tests/functional/d` dir.

Current failures in `pre-commit` are indeed from broken `main`.

**Comment by omarandlorraine** [comment]
Oh, thanks for taking a look. 

That file didn't exist here; that's why I couldn't see it! It's new since PR#6360, which was merged in recently. So I've done the necessaries, let's see if the tests pass this time.

**Comment by omarandlorraine** [comment]
Okay, now some tests for W1203 are failing. I'm certain I didn't touch W1203. 

Am I missing a step here or something?

**Comment by DanielNoord** [comment]
@omarandlorraine I think you were just a little too quick to merge `main`. There is actually one more commit on `main` that is need to fix it. That should fix it!

**Comment by omarandlorraine** [comment]
Okay, let's try this.

There has been so much toing and froing and pulling and rebasing for this simple pull request. Because someone else is working on the main branch. Is this normal?

**Comment by DanielNoord** [comment]
> Okay, let's try this.
> 
> There has been so much toing and froing and pulling and rebasing for this simple pull request. Because someone else is working on the main branch. Is this normal?

No, this not normal. You just caught in 24h of extreme unluckiness. 
First we broke one our linter on `main` because two connected PRs passed all checks on their own, but didn't pass when we committed them to `main` together.
Then, while we were in the process of fixing that linting message we broke our test suite with another set of connected PRs that passed on their own.

Both issues have now been resolved and there shouldn't be any need to merge `main` into this branch to get the CI to pass from now on. Just extreme unlucky... Sorry about that! 😄

**Comment by Pierre-Sassoulas** [comment]
Yeah.. it's the first time main was broken since... maybe august 2021, 8 months ago (https://github.com/PyCQA/pylint/pull/4801), sorry for the unusually bad experience.

**Comment by omarandlorraine** [comment]
Oh, that's good. I'll keep on contributing in the future then 🙂 I can see that the checks are still failing though. But I can't see the logs, maybe because I'm on my phone right now. 

I'll have another look tomorrow.

**Review by Pierre-Sassoulas — APPROVED** [review]
Took the liberty to rebase because the intense activity on main made the merge strategy problematic.

**Review by DanielNoord — APPROVED** [review]

**Comment by DanielNoord** [comment]
@Pierre-Sassoulas I'm seeing a lot of different emails in the `commit` messages. Could you give some guidance on how to add those to the contributor list?

**Comment by Pierre-Sassoulas** [comment]
One of the adresse was "you@example.com" so I removed it. But it's possible to create alias in https://github.com/PyCQA/pylint/blob/main/script/.contributors_aliases.json otherwise.

Thank you for your contribution @omarandlorraine !

**Comment by omarandlorraine** [comment]
You're welcome!

Thanks for the help merging and closing it.

**Pull request merged**
